backend/test/modules/doubles-tournament/core/application/CallingEnqueuedMatches.spec.ts

```diff
@@ -0,0 +1,171 @@
+import { CommandResult } from '../../../../../src/shared/core/application/command/CommandResult';
+import { StartMatch } from '../../../../../src/modules/match-module/core/application/command/StartMatch';
+import { TestModuleCore } from '../../../../test-support/shared/core/TestModuleCore';
+
+describe('Calling Enqueued Matches', () => {
+  const currentTime = new Date();
+  const tournamentId = 'TournamentId';
+  const matchNumber = 3;
+  const team1Id = 'Team1Id';
+  const team2Id = 'Team2Id';
+  const matchNumber2 = 5;
+  const team1Id2 = 'team1Id2';
+  const team2Id2 = 'team2Id2';
+  const enqueueMatch = new EnqueueMatch({
+    tournamentId: tournamentId,
+    matchNumber: matchNumber,
+    team1Id: team1Id,
+    team2Id: team2Id,
+  });
+  const enqueueMatch2 = new EnqueueMatch({
+    tournamentId: tournamentId,
+    matchNumber: matchNumber2,
+    team1Id: team1Id2,
+    team2Id: team2Id2,
+  });
+  const tableNumber = 1;
+  const tableNumber2 = 2;
+  const table1Booked = new TournamentTableWasBooked({ occurredAt: currentTime, tournamentId: tournamentId, tableNumber: tableNumber });
+  const table1Released = new TournamentTableWasReleased({ occurredAt: currentTime, tournamentId: tournamentId, tableNumber: tableNumber });
+  const table2Booked = new TournamentTableWasBooked({ occurredAt: currentTime, tournamentId: tournamentId, tableNumber: tableNumber2 });
+  const table2Released = new TournamentTableWasReleased({
+    occurredAt: currentTime,
+    tournamentId: tournamentId,
+    tableNumber: tableNumber2,
+  });
+  let spy: jest.SpyInstance;
+  let doublesTournament: TestModuleCore;
+
+  beforeEach(() => {
+    const commandBus: CommandBus = new InMemoryCommandBus();
+    spy = jest.spyOn(commandBus, 'execute');
+    const entityIdGen = FromListIdGeneratorStub([team1Id, team2Id, team1Id2, team2Id2]);
+    doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+    const alwaysSuccessStartMatchCommandHandler: CommandHandler<StartMatch> = {
+      async execute(command: StartMatch): Promise<CommandResult> {
+        return CommandResult.success();
+      },
+    };
+    commandBus.registerHandler(StartMatch, alwaysSuccessStartMatchCommandHandler);
+    const alwaysSuccessBookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.success();
+      },
+    };
+    commandBus.registerHandler(BookTournamentTable, alwaysSuccessBookTournamentTableCommandHandler);
+  });
+
+  it('When matches were enqueued and only one table was released then call the match with lower matchNumber', async () => {
+    //Given
+    await doublesTournament.executeCommand(enqueueMatch2);
+    await doublesTournament.executeCommand(enqueueMatch);
+    spy.mockClear();
+
+    //When
+    await doublesTournament.publishEvent(table1Booked);
+    await doublesTournament.publishEvent(table2Released);
```

Comment:
Pytanie:
Siedzimy z Tomkiem nad jego testami i tutaj się zastanawiamy nad tym czemu moduł DoubleTournament publikuje tutaj event, a nie TournamentTable? 
![image](https://user-images.githubusercontent.com/31566345/112988159-d2d69480-9163-11eb-934b-715798cf9ec9.png)

Reply:
Nie znam się, ale skoro ja handluję u siebie dany event, to musi być dodany do moduleCore, więc DoubleTournament go rozpoznaje, mimo że pochodzi z innego modułu, a skoro tak, to chyba mogę go sobie opublikować. Choć logiczniej byłoby może gdyby faktycznie był publikowany przez inny moduł.

Reply:
Zmienione :)